File: frontend/sync.py
import re

with open('src/main/webapp/views/stack.html', 'r', encoding='utf-8') as f:
    stack_html = f.read()

with open('src/main/webapp/views/lider-proyecto.html', 'r', encoding='utf-8') as f:
    lider_html = f.read()

# Extract tech categories from stack.html
start_idx = stack_html.find('<div class="tech-category">')
end_idx = stack_html.find('<div class="action-footer">')
stack_section = stack_html[start_idx:end_idx].strip()

# Replace trailing div from cards-grid if included
if stack_section.endswith('</div>'):
    stack_section = stack_section[:-6].strip()

# Change class="tech-tag" to type="button" class="stack-tag"
stack_section = stack_section.replace('class="tech-tag"', 'type="button" class="stack-tag"')

# Now insert into lider-proyecto.html
# We find the start of the first <div class="tech-category"> in lider-proyecto.html
lider_start = lider_html.find('<div class="tech-category">')

# We find the end where <input type="hidden" name="stack" id="stack-hidden" required> is
lider_end = lider_html.find('<input type="hidden" name="stack" id="stack-hidden" required>')

if lider_start != -1 and lider_end != -1:
    final_html = lider_html[:lider_start] + stack_section + '\n  ' + lider_html[lider_end:]
    with open('src/main/webapp/views/lider-proyecto.html', 'w', encoding='utf-8') as f:
        f.write(final_html)
    print("Success")
else:
    print("Failed")

File: frontend/sync.js
const fs = require('fs');
let stackHtml = fs.readFileSync('src/main/webapp/views/stack.html', 'utf8');
let liderHtml = fs.readFileSync('src/main/webapp/views/lider-proyecto.html', 'utf8');

const stackStart = stackHtml.indexOf('<div class="tech-category">');
const stackEnd = stackHtml.indexOf('</div>\n\n            <div class="action-footer">');
let newContent = stackHtml.substring(stackStart, stackEnd).trim();

// Replace tech-tag with stack-tag for lider-proyecto
newContent = newContent.replace(/class="tech-tag"/g, 'type="button" class="stack-tag"');

const liderStart = liderHtml.indexOf('<div class="stack-picker" id="stack-picker">') + '<div class="stack-picker" id="stack-picker">'.length;
const liderEnd = liderHtml.indexOf('  </div>\n  <input type="hidden" name="stack" id="stack-hidden" required>');

if (liderStart !== -1 && liderEnd !== -1) {
    const finalHtml = liderHtml.substring(0, liderStart) + '\n    \n    ' + newContent + '\n\n' + liderHtml.substring(liderEnd);
    fs.writeFileSync('src/main/webapp/views/lider-proyecto.html', finalHtml, 'utf8');
    console.log('Successfully updated lider-proyecto.html');
} else {
    console.log('Failed to find boundaries in lider-proyecto.html');
}
